src/pybind/mgr/dashboard/tests/test_cluster_upgrade.py

```diff
@@ -0,0 +1,62 @@
+from ..controllers.cluster_upgrade import ClusterUpgrade
+from ..tests import ControllerTestCase, patch_orch
+from ..tools import NotificationQueue, TaskManager
+
+
+class ClusterUpgradeControllerTest(ControllerTestCase):
+    URL_CLUSTER_UPGRADE = '/api/cluster/upgrade'
+
+    @classmethod
+    def setup_server(cls):
+        NotificationQueue.start_queue()
+        TaskManager.init()
+        cls.setup_controllers([ClusterUpgrade])
+
+    @classmethod
+    def tearDownClass(cls):
+        NotificationQueue.stop()
+
+    def test_upgrade_list(self):
+        result = ['17.1.0', '16.2.7', '16.2.6', '16.2.5', '16.1.4', '16.1.3']
+        with patch_orch(True) as fake_client:
+            fake_client.upgrades.list.return_value = result
+            self._get('{}?image=quay.io/ceph/ceph:v16.1.0&tags=False&show_all_versions=False'
+                      .format(self.URL_CLUSTER_UPGRADE))
+            self.assertStatus(200)
+            self.assertJsonBody(result)
+
+    def test_start_upgrade(self):
+        msg = "Initiating upgrade to 17.2.6"
+        with patch_orch(True) as fake_client:
+            fake_client.upgrades.start.return_value = msg
+            payload = {
+                'image': 'quay.io/ceph/ceph',
+                'version': '17.2.6'
```

Comment:
we don't need both `image` and `version` to be provided. Either we should provide an `image` or we should provide the `version`